### PR TITLE
Data catalog Tag module

### PR DESCRIPTION
**Pull request by lcaggio**
Data catalog Tag module. This is a simple module. The main reason for creating a module is the Tag factory.

---
**Checklist**

I applicable, I acknowledge that I have:
- [x] Read the [contributing guide](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md)
- [x] Ran `terraform fmt` on all modified files
- [x] Regenerated the relevant README.md files using [`tools/tfdoc.py`](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md#fabric-tools)
- [x] Made sure all relevant tests pass

### CODE REVIEW

**Comment by thinhha** [comment]
This looks great! Thanks Lorenzo.

**Review by juliocc — APPROVED** [review]

File: modules/data-catalog-tag/README.md [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+
+```hcl
+module "data-catalog-tag" {
+  source = "./fabric/modules/data-catalog-tag"
+  tags = {
+    "landing/countries" = {
+      project_id = "project-data-product"
+      parent     = "projects/project-data-product/datasets/landing/tables/countries"
+      column     = "country"
+      location   = "europe-west-1"
```
Comment (wiktorn) [inline modules/data-catalog-tag/README.md:61]:
```suggestion
      location   = "europe-west1"
```
?

**Review by wiktorn — APPROVED** [review]
I wonder about actual usage. @lcaggio based on your experience - is it multiple templates applied to single parent, or rather one template per parent?

Also, can template location can be different than tag location?

I wonder about remodeling this structure a bit, to reduce boilerplate code, like for example if I have multiple `column` on one `parent` to tag

**Comment by lcaggio** [comment]
Thanks @wiktorn. From my experience, multiple Tag can be assigned to a resource (e.g. multiple tag on a dataset, less common for columns).

I think location can be different, but I am going to use this module in 2 customers really soon and I will check the structure of the module with reality of customer needs. I will update the module accordingly.

**Pull request merged**